src/java/org/apache/cassandra/index/sai/analyzer/EqualsSupport.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.analyzer;
+
+import java.util.Arrays;
+import java.util.Map;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import org.apache.cassandra.exceptions.InvalidRequestException;
+
+/**
+ * Index config property for defining the behaviour of the equals operator (=) when the index is analyzed.
+ * </p>
+ * Analyzers transform the indexed value, so EQ queries using an analyzed index can return results diferent to those of
+ * an equivalent query without indexes. Having EQ queries returning different results depending on if/how the column is
+ * indexed can be confussing for users, so probably the safest approach is to reject EQ queries on analyzed indexes, and
+ * let users use the analuzer matches operator (:) instead. However, for backwards compatibility reasons, we should
+ * allow users to let equality queries behave same as match queries through this index config property. We use an enum
+ * value rather than a boolean to allow for future extensions.
+ */
+public class EqualsSupport
+{
+    public static final String OPTION = "equals_behaviour_when_analyzed";
+    public static final Value DEFAULT = Value.MATCH; // default to : behaviour for backwards compatibility
```

Comment:
That's a great idea. We want to change how users write `SELECT` queries, so I have added a client warning there. 

I'm not so fond of adding a warning on `CREATE INDEX` because we might be punishing users who use the defaults and query analyzed columns with `:`, which is the ideal behaviour we want to have in the future. 

Also, it can be a bad user experience if we welcome users with a warning the first time they create an index, asking them to remove it and create it again with the obscure `equals_behaviour_when_analyzed` config option.

Reply:
`Also, it can be a bad user experience if we welcome users with a warning the first time they create an index, asking them to remove it and create it again with the obscure equals_behaviour_when_analyzed config option.`
I see what you mean; technically, if they will have the same behavior anyway when using `:`, it doesn't matter. And it is not that we will break those indexes in the future, I assume. We were even discussing the `ALTER INDEX` Fine with emitting it only for `SELECT` queries.